### PR TITLE
Add github app config information

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

**Comment by gewenyu99** [comment]
localhost_2080_docs_configuration

**Review by vermakhushboo — COMMENTED** [review]
Let's also add how to configure webhook url. 
`https://<ngrok url>/v1/vcs/github/events`

**Comment by vermakhushboo** [comment]
Maybe this detail as well: Everytime user updates their GitHub app's setting, they need to go to their app installation page and approve the prompt - `GitHub app is asking for updated permissions`

File: app/views/docs/configuration.phtml [hunk 1]
```diff
@@ -22,10 +22,128 @@
...
+    </tr>
+    <tr>
+        <td><code>http://[HOSTNAME_OR_IP]/v1/account/sessions/oauth2/callback/github/console</code></td>
+    </tr>
+    </tbody>
+</table>
+
+<h3><a href="#post-install" id="post-install">Post Installation</a></h3>
+<p>
+    Check the <b> Redirect on update</b> box under the <b>Post installation</b> section.
```
Comment (stnguyen90) [inline app/views/docs/configuration.phtml:57]:
```suggestion
    Check the <b>Redirect on update</b> box under the <b>Post installation</b> section.
```

File: app/views/docs/configuration.phtml [hunk 2]
```diff
@@ -22,10 +22,128 @@
...
+    </tr>
+    <tr>
+        <td>Pull requests</td>
+        <td>Read and write</td>
+    </tr>
+    <tr>
+        <td>Webhooks</td>
+        <td>Read and write</td>
+    </tr>
+    </tbody>
```
Comment (stnguyen90) [inline app/views/docs/configuration.phtml:104]: indentation:

```suggestion
    <thead>
        <tr>
            <td>Permission</td>
            <td>Access</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Administration</td>
            <td>Read and write</td>
        </tr>
        <tr>
            <td>Checks</td>
            <td>Read and write.</td>
        </tr>
        <tr>
            <td>Commit Statuses</td>
            <td>Read and write</td>
        </tr>
        <tr>
            <td>Contents</td>
            <td>Read and write</td>
        </tr>
        <tr>
            <td>Issues</td>
            <td>Read and write</td>
        </tr>
        <tr>
            <td>Metadata</td>
            <td>Read-only</td>
        </tr>
        <tr>
            <td>Pull requests</td>
            <td>Read and write</td>
        </tr>
        <tr>
            <td>Webhooks</td>
            <td>Read and write</td>
        </tr>
    </tbody>
```

File: app/views/docs/configuration.phtml [hunk 3]
```diff
@@ -22,10 +22,128 @@
...
+        <td>Permission</td>
+        <td>Access</td>
+    </tr>
+    </thead>
+    <tbody>
+    <tr>
+        <td>Email address</td>
+        <td>Read-only</td>
+        </tr>
+    </tbody>
```
Comment (stnguyen90) [inline app/views/docs/configuration.phtml:123]: indentation:

```suggestion
    <thead>
        <tr>
            <td>Permission</td>
            <td>Access</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Email address</td>
            <td>Read-only</td>
        </tr>
    </tbody>
```

File: app/views/docs/configuration.phtml [hunk 4]
```diff
@@ -22,10 +22,128 @@
...
+        <td>Event</td>
+        <td></td>
+        <td></td>
+    </tr>
+    </thead>
+    <tbody>
+    <tr>
+        <td>Pull request</td> <td>Push</td> <td></td>
+    </tr>
+    </tbody>
```
Comment (stnguyen90) [inline app/views/docs/configuration.phtml:142]: indentation:

```suggestion
    <thead>
        <tr>
            <td>Event</td>
            <td></td>
            <td></td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Pull request</td>
            <td>Push</td>
            <td></td>
        </tr>
    </tbody>
```

File: app/views/docs/functions-develop.phtml [hunk 5]
```diff
@@ -2637,32 +2637,32 @@ public class Main {
...
             The response object has been updated.
             You can now specify headers, as well as use new methods like return redirects or empty responses.
-            <a href="docs/functions-develop#response">Learn about response.</a>
+            <a href="/docs/functions-develop#response">Learn about response.</a>
         </li>
         <li>
             Now, you <b>must return a response</b> such as <code>return context.res.send("")</code>.
             This prevents confusing errors when functions are terminated prematurely before a response is sent.
-            <a href="docs/functions-develop#response">Learn about response.</a>
+            <a href="/docs/functions-develop#response">Learn about response.</a>
```
Comment (stnguyen90) [inline app/views/docs/functions-develop.phtml:2665]: why not use a fragment for these?

```suggestion
            <a href="#response">Learn about response.</a>
```

**Review by stnguyen90 — APPROVED** [review]
Minor formatting suggestions.

I didn't read the content, though.

**Comment by gewenyu99** [comment]
localhost_2080_docs_configuration (3)

**Review by vermakhushboo — APPROVED** [review]

File: app/views/docs/configuration.phtml [hunk 6]
```diff
@@ -64,6 +197,14 @@
 
 <p><a href="/docs/environment-variables"><i class="icon-angle-circled-right margin-start-negative-tiny margin-end-tiny"></i> Learn more about environment variables</a></p>
 
+<h3><a href="#existing-apps" id="existing-apps">Update Existing GitHub Apps</a></h3>
+<p>
+    If you're editing existing GitHub apps, there are additional steps.
+    Everytime you update your GitHub app's setting, GitHub will prompt you to review the changes.
```
Comment (gewenyu99) [inline app/views/docs/configuration.phtml:203]:
```suggestion
    Every time you update your GitHub app's permissions, GitHub will prompt you to review the changes.
```

File: app/views/docs/configuration.phtml [hunk 7]
```diff
@@ -64,6 +197,14 @@
 
 <p><a href="/docs/environment-variables"><i class="icon-angle-circled-right margin-start-negative-tiny margin-end-tiny"></i> Learn more about environment variables</a></p>
 
+<h3><a href="#existing-apps" id="existing-apps">Update Existing GitHub Apps</a></h3>
+<p>
+    If you're editing existing GitHub apps, there are additional steps.
```
Comment (gewenyu99) [inline app/views/docs/configuration.phtml:202]:
```suggestion
    There are additional steps if you're updating permissions in existing GitHub apps.
```

File: app/views/docs/configuration.phtml [hunk 8]
```diff
@@ -64,6 +197,14 @@
...
+    Every time you update your GitHub app's permissions, GitHub will prompt you to review the changes.
+    You will receive an email and you'll find a prompt under <b>Settings</b> > <b>Integrations</b> > <b>Applications</b> > <b>Installed GitHub Apps</b> > find your GitHub app > <b>Configure</b>.
+    You need accept the change so they're applied to your app.
```
Comment (gewenyu99) [inline app/views/docs/configuration.phtml:205]:
```suggestion
    You need to accept the new permissions so they're applied to your app.
```

File: app/views/docs/configuration.phtml [hunk 9]
```diff
@@ -22,10 +22,143 @@
...
     Appwrite supports automatic deployments through Git integration. 
-    In order for Appwrite to access your repos, you must configure a GitHub app to enable this integration.
+    In order for Appwrite to access your repos, you must <a href="https://docs.github.com/en/apps/creating-github-apps/about-creating-github-apps/about-creating-github-apps" target="_blank" rel="noopener">create a GitHub app</a> to enable this integration.
+    The GitHub app requires the following configurations.
 </p>
 
-<p>You'll have to configure the following environment variables.</p>
+<p>
+    For automatic Git deployment to work, Appwrite needs to receive communication from GitHub, this means your Appwrite project must be accessible on the internet.
+    If you're running on <code>localhost</code>, you need to run a proxy like <b>ngrok</b>.
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:31]: can we add a hyperlink to the ngrok website

File: app/views/docs/configuration.phtml [hunk 10]
```diff
@@ -22,10 +22,143 @@
 <h2><a href="#git" id="git">Configure GitHub App</a></h2>
```
[the remark this answers is not in the document]
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:22]: Screenshot 2023-08-31 at 4 07 08 PM
You don't need this in cloud

[the remark this answers is not in the document]
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:22]: It's the big config page for self-hosted

File: app/views/docs/configuration.phtml [hunk 11]
```diff
@@ -22,10 +22,143 @@
...
+        </tr>
+        <tr>
+            <td><code>http://[HOSTNAME_OR_IP]/v1/account/sessions/oauth2/callback/github/console</code></td>
+        </tr>
+    </tbody>
+</table>
+
+<h3><a href="#post-install" id="post-install">Post Installation</a></h3>
+<p>
+    Check the <b>Redirect on update</b> box under the <b>Post installation</b> section.
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:57]: This is lacking context as to which page the developer needs to check this option in. We should attach a screenshot or provide a link to the page in the previous section. 

Something like "Head to https://github.com/path/to/create/app and do ... "

Comment (gewenyu99) [reply app/views/docs/configuration.phtml:57]: Oh your right, the settings link is static good call

File: app/views/docs/configuration.phtml [hunk 12]
```diff
@@ -22,10 +22,143 @@
...
+    <thead>
+        <tr>
+            <td>Permission</td>
+            <td>Access</td>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>Administration</td>
+            <td>Read and write</td>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:87]: weren't we going to explain why we need each of these permissions?

File: app/views/docs/configuration.phtml [hunk 13]
```diff
@@ -22,10 +22,143 @@
...
+<h3><a href="#subscribe-events" id="subscribe-events">Subscribe to Events</a></h3>
+<p>
+    Select these under the <b>Subscribe to events</b> dropdown.
+</p>
+<table class="full">
+    <thead>
+        <tr>
+            <td>Event</td>
+            <td></td>
+            <td></td>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:148]: are these cells left empty intentionally ?

File: app/views/docs/configuration.phtml (hunk 12 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:87]: We didn't have time

File: app/views/docs/configuration.phtml (hunk 13 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:148]: yes, for spacing

Comment (gewenyu99) [reply app/views/docs/configuration.phtml:148]: Peep the screenshot

File: app/views/docs/configuration.phtml (hunk 12 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:87]: Maybe in the future, when I can pull Khushboo and matej aside.

File: app/views/docs/configuration.phtml (hunk 9 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml]: Linked

File: app/views/docs/configuration.phtml (hunk 13 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:148]: Screenshot 2023-08-31 at 4 13 51 PM

File: app/views/docs/configuration.phtml [hunk 14]
```diff
@@ -80,18 +221,18 @@
...
 
-<p>Appwrite supports AWS S3, Digital Ocean Spaces, Backblaze, Akamai Object Storage, and Wasabi as storage adaptors. Some of these services can be self-hosted, just like Appwrite.</p>
+<p>Appwrite supports AWS S3, Digital Ocean Spaces, Backblaze, Akamai Object Storage, and Wasabi as storage adapters. Some of these services can be self-hosted, just like Appwrite.</p>
 
-<p>You can select which storage adaptor to use by setting the <code>_APP_STORAGE_DEVICE</code> environment variable. Valid values are <code>local</code>, <code>s3</code>, <code>dospaces</code>, <code>backblaze</code>, <code>linode</code> and <code>wasabi</code>. Each storage adaptor requires its own set of additional environment variables to configure.</p>
+<p>You can select which storage adaptor to use by setting the <code>_APP_STORAGE_DEVICE</code> environment variable. Valid values are <code>local</code>, <code>s3</code>, <code>dospaces</code>, <code>backblaze</code>, <code>linode</code>, and <code>wasabi</code>. Each storage adaptor requires its own set of additional environment variables to configure.</p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:231]:
```suggestion
<p>You can select which storage adapter to use by setting the <code>_APP_STORAGE_DEVICE</code> environment variable. Valid values are <code>local</code>, <code>s3</code>, <code>dospaces</code>, <code>backblaze</code>, <code>linode</code>, and <code>wasabi</code>. Each storage adaptor requires its own set of additional environment variables to configure.</p>
```

File: app/views/docs/configuration.phtml (hunk 11 above)
Comment (christyjacob4) [reply app/views/docs/configuration.phtml:57]: Alright then 👍

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**